xradar/io/backends/nexrad_level2.py

```diff
@@ -1253,8 +1252,11 @@ def _getitem(self, key):
         except KeyError:
             self.datastore.root.get_data(self.group, self.name)
             data = self.datastore.ds["sweep_data"][self.name]["data"]
+        # see 3.2.4.17.6 Table XVII-I Data Moment Characteristics and Conversion for Data Names
         if self.name == "PHI":
```

Comment:
This actually only represents the latest ICD's https://www.roc.noaa.gov/interface-control-documents.php. How do we check for data based on earlier ICD's? We could just add a table with the versions and the differences.



We would also need to mask out 0 (indicates received signal is below threshold) and 1 (
indicates range folded data).

Reply:
@syedhamidali This will only read PHI and ZDR correctly for the latest ICD based radar data. I'll add a check for word_size for the ZDR to only use 11 bit for 2 byte data. Any other suggestions from your side, how to tackle this properly?

Reply:
I think it reads as it was before

![image](https://github.com/user-attachments/assets/57546619-3633-4a16-8fbd-14605113e0ff)

Reply:
OK, I meant, if we have older data with 1byte ZDR, that might not read correct.

Reply:
When you are on it, could you check if PHIDP does make sense?

Reply:
Looks good to me

![image](https://github.com/user-attachments/assets/1713a540-7214-445d-9c46-8784cd07364a)

Reply:
and this looks also fine

![image](https://github.com/user-attachments/assets/047f52b4-46d0-48c4-99e4-cb1e9f74c22a)

Reply:
OK, great! Thanks for checking. I think with the current 2byte check, we are on the safe side. All other enhancements/fixes for nexrad should be tackled when working on #40 and #163.